libkbfs/folder_branch_ops.go

```diff
@@ -1914,6 +1915,19 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 
 	var mdID MdID
 
+	if err = EnableDelayedCancellationWithGracePeriod(
```

Comment:
A comment here would be good, explaining that this acts somewhat like a critical section, to help us know for sure whether the MD put made it to the server or not even when the operation is cancelled.  This is useful even when `excl != WithExcl`, since it helps us avoid self-conflicts.  But it comes at the cost of a less-responsive ctrl-C once the operation has reached this point.